### PR TITLE
Generate node parameters using generate_parameter_library

### PR DESCRIPTION
**Pull request by ghost**
### Description

- generate node parameters using `generate_parameter_library` in `husarion_ugv_battery`, `husarion_ugv_lights` and `husarion_ugv_manager` packages

### Requirements

- [x] Code style guidelines followed
- [x] Documentation updated

### Tests 🧪

- [ ] Robot
- [ ] Container
- [x] Simulation

### CODE REVIEW

File: husarion_ugv_battery/src/battery_driver_node.cpp [hunk 1]
```diff
@@ -75,22 +76,17 @@ void BatteryDriverNode::InitializeWithADCBattery()
...
 
   adc0_reader_ = std::make_shared<ADCDataReader>(adc0_device_path);
   adc1_reader_ = std::make_shared<ADCDataReader>(adc1_device_path);
 
   const ADCBatteryParams battery_params = {
-    static_cast<std::size_t>(this->get_parameter("ma_window_len/voltage").as_int()),
-    static_cast<std::size_t>(this->get_parameter("ma_window_len/current").as_int()),
-    static_cast<std::size_t>(this->get_parameter("adc/ma_window_len/temp").as_int()),
-    static_cast<std::size_t>(this->get_parameter("adc/ma_window_len/charge").as_int()),
+    static_cast<std::size_t>(this->params_.ma_window_len.voltage),
```
Comment (delihus) [inline husarion_ugv_battery/src/battery_driver_node.cpp:86]: Do they still need to be casted?

File: husarion_ugv_lights/config/lights_controller_parameters.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+lights_controller:
+  animations_config_path:
+    type: string
+    default_value: ""
+    description: Path to a YAML file with a description of led configuration. This file includes definition of robot panels, virtual segments and default animations.
```
Comment (delihus) [inline husarion_ugv_lights/config/lights_controller_parameters.yaml]: In the read me was `LEDs`
```suggestion
    description: Path to a YAML file with a description of LED configuration. This file includes definition of robot panels, virtual segments and default animations.
```

File: husarion_ugv_lights/src/lights_controller_node.cpp [hunk 3]
```diff
@@ -48,13 +50,13 @@ LightsControllerNode::LightsControllerNode(const rclcpp::NodeOptions & options)
...
+  this->param_listener_ =
+    std::make_shared<lights_controller::ParamListener>(this->get_node_parameters_interface());
+  this->params_ = this->param_listener_->get_params();
 
-  const auto animations_config_path = this->get_parameter("animations_config_path").as_string();
-  const auto user_led_animations_path = this->get_parameter("user_led_animations_path").as_string();
-  const float controller_freq = this->get_parameter("controller_freq").as_double();
+  const auto animations_config_path = this->params_.animations_config_path;
+  const auto user_led_animations_path = this->params_.user_led_animations_path;
+  const float controller_freq = this->params_.controller_frequency;
```
Comment (delihus) [inline husarion_ugv_lights/src/lights_controller_node.cpp]: It looks like conversion from double to float. Does the controller needs floats?

File: husarion_ugv_manager/config/lights_manager_parameters.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      type: double
+      default_value: 0.1
+      description: This parameter defines the minimum change in battery percentage required to trigger an update in the battery charging animation.
+      validation: { gt<>: 0.0 }
+
+    anim_period:
+      critical:
+        type: double
+        default_value: 15.0
+        description: Time in seconds to wait before repeating animation, indicating a critical Battery state.
```
Comment (delihus) [inline husarion_ugv_manager/config/lights_manager_parameters.yaml]:
```suggestion
        description: Time in seconds to wait before repeating animation, indicating a critical battery state.
```

File: husarion_ugv_manager/config/lights_manager_parameters.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        validation: { gt<>: 0.0 }
+
+      low:
+        type: double
+        default_value: 30.0
+        description: Time in seconds to wait before repeating the animation, indicating a low Battery state.
```
Comment (delihus) [inline husarion_ugv_manager/config/lights_manager_parameters.yaml]:
```suggestion
        description: Time in seconds to wait before repeating the animation, indicating a low battery state.
```

File: husarion_ugv_manager/config/lights_manager_parameters.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        validation: { gt<>: 0.0 }
+
+    percent:
+      window_len:
+        type: int
+        default_value: 6
+        description: Moving average window length used to smooth out Battery percentage readings.
```
Comment (delihus) [inline husarion_ugv_manager/config/lights_manager_parameters.yaml]:
```suggestion
        description: Moving average window length used to smooth out battery percentage readings.
```

File: husarion_ugv_manager/config/lights_manager_parameters.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+        validation: { gt<>: 0 }
+
+      threshold:
+        critical:
+          type: double
+          default_value: 0.1
+          description: If the Battery percentage drops below this value, an animation indicating a critical Battery state will start being displayed.
```
Comment (delihus) [inline husarion_ugv_manager/config/lights_manager_parameters.yaml]:
```suggestion
          description: If the battery percentage drops below this value, an animation indicating a critical battery state will start being displayed.
```

File: husarion_ugv_manager/config/lights_manager_parameters.yaml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+          validation: { bounds<>: [0.0, 1.0] }
+
+        low:
+          type: double
+          default_value: 0.4
+          description: If the Battery percentage drops below this value, the animation indicating a low Battery state will start being displayed.
```
Comment (delihus) [inline husarion_ugv_manager/config/lights_manager_parameters.yaml]:
```suggestion
          description: If the battery percentage drops below this value, the animation indicating a low battery state will start being displayed.
```

File: husarion_ugv_manager/config/lights_manager_parameters.yaml [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  bt_project_path:
+    type: string
+    default_value: ""
+    description: Path to a BehaviorTree project.
+    validation: { not_empty<> }
+
+  plugin_libs:
+    type: string_array
+    default_value: []
+    description: List with names of plugins that are used in the BT project.
```
Comment (delihus) [inline husarion_ugv_manager/config/lights_manager_parameters.yaml]:
```suggestion
    description: A list with names of plugins that are used in the BehaviorTree project.
```

File: husarion_ugv_manager/config/lights_manager_parameters.yaml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+
+  ros_communication_timeout:
+    availability:
+      type: double
+      default_value: 1.0
+      description: Timeout in seconds to wait for a service/action while initializing BT node.
```
Comment (delihus) [inline husarion_ugv_manager/config/lights_manager_parameters.yaml]:
```suggestion
      description: Timeout in seconds to wait for a service/action while initializing a BehaviorTree node.
```

File: husarion_ugv_manager/config/lights_manager_parameters.yaml [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    response:
+      type: double
+      default_value: 1.0
+      description: Timeout in seconds to receive a service/action response after call.
+      validation: { gt<>: 0.0 }
+
+  ros_plugin_libs:
+    type: string_array
+    default_value: []
+    description: List with names of ROS plugins that are used in a BT project.
```
Comment (delihus) [inline husarion_ugv_manager/config/lights_manager_parameters.yaml]:
```suggestion
    description: A list with names of ROS plugins that are used in a BehaviorTree project.
```

File: husarion_ugv_manager/config/safety_manager_parameters.yaml [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+      fan_off:
+        type: double
+        default_value: 35.0
+        description: Temperature in degrees Celsius of any drivers below which the fan is turned off.
+        validation: { bounds<>: [-20.0, 100.0] }
+
+      fan_on:
+        type: double
+        default_value: 45.0
+        description: Temperature in degrees Celsius* of any drivers above which the fan is turned on.
```
Comment (delihus) [inline husarion_ugv_manager/config/safety_manager_parameters.yaml]:
```suggestion
        description: Temperature in degrees Celsius of any drivers above which the fan is turned on.
```

File: husarion_ugv_manager/config/safety_manager_parameters.yaml [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+  fan_turn_off_timeout:
+    type: double
+    default_value: 60.0
+    description: Minimal time in seconds, after which the fan may be turned off.
+    validation: { gt<>: 0.0 }
+
+  plugin_libs:
+    type: string_array
+    default_value: []
+    description: List with names of plugins that are used in the BT project.
```
Comment (delihus) [inline husarion_ugv_manager/config/safety_manager_parameters.yaml]:
```suggestion
    description: A list with names of plugins that are used in the BehaviorTree project.
```

File: husarion_ugv_manager/config/safety_manager_parameters.yaml [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+  ros_communication_timeout:
+    availability:
+      type: double
+      default_value: 1.0
+      description: Timeout seconds to wait for a service/action while initializing BT node.
```
Comment (delihus) [inline husarion_ugv_manager/config/safety_manager_parameters.yaml]:
```suggestion
      description: Timeout seconds to wait for a service/action while initializing BehaviorTree node.
```

File: husarion_ugv_manager/config/safety_manager_parameters.yaml [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+    response:
+      type: double
+      default_value: 1.0
+      description: Timeout seconds to receive a service/action response after call.
+      validation: { gt<>: 0.0 }
+
+  ros_plugin_libs:
+    type: string_array
+    default_value: []
+    description: List with names of ROS plugins that are used in a BT project.
```
Comment (delihus) [inline husarion_ugv_manager/config/safety_manager_parameters.yaml]:
```suggestion
    description: A list with names of ROS plugins that are used in a BehaviorTree project.
```

File: husarion_ugv_manager/src/lights_manager_node.cpp [hunk 16]
```diff
@@ -130,15 +106,11 @@ void LightsManagerNode::RegisterBehaviorTree()
...
-    this->get_parameter("battery.percent.threshold.critical").as_double();
-  const float low_battery_anim_period =
-    this->get_parameter("battery.animation_period.low").as_double();
-  const float low_battery_threshold_percent =
-    this->get_parameter("battery.percent.threshold.low").as_double();
+  update_charging_anim_step_ = this->params_.battery.charging_anim_step;
+  const float critical_battery_anim_period = this->params_.battery.anim_period.critical;
+  const float critical_battery_threshold_percent = this->params_.battery.percent.threshold.critical;
+  const float low_battery_anim_period = this->params_.battery.anim_period.low;
+  const float low_battery_threshold_percent = this->params_.battery.percent.threshold.low;
```
Comment (delihus) [inline husarion_ugv_manager/src/lights_manager_node.cpp]: Conversion from double to float. I would change this to auto or double.

File: husarion_ugv_manager/src/safety_manager_node.cpp [hunk 17]
```diff
@@ -96,7 +98,7 @@ void SafetyManagerNode::Initialize()
   //   Timers
   // -------------------------------
 
-  const float timer_freq = this->get_parameter("timer_frequency").as_double();
+  const float timer_freq = this->params_.timer_frequency;
```
Comment (delihus) [inline husarion_ugv_manager/src/safety_manager_node.cpp]: Conversion from double to float

File: husarion_ugv_manager/src/safety_manager_node.cpp [hunk 18]
```diff
@@ -160,11 +134,11 @@ void SafetyManagerNode::RegisterBehaviorTree()
 
 std::map<std::string, std::any> SafetyManagerNode::CreateSafetyInitialBlackboard()
 {
-  const float cpu_fan_on_temp = this->get_parameter("cpu.temp.fan_on").as_double();
-  const float cpu_fan_off_temp = this->get_parameter("cpu.temp.fan_off").as_double();
-  const float driver_fan_on_temp = this->get_parameter("driver.temp.fan_on").as_double();
-  const float driver_fan_off_temp = this->get_parameter("driver.temp.fan_off").as_double();
-  const float fan_turn_off_timeout = this->get_parameter("fan_turn_off_timeout").as_double();
+  const float cpu_fan_on_temp = this->params_.cpu.temp.fan_on;
```
Comment (delihus) [inline husarion_ugv_manager/src/safety_manager_node.cpp]: Also double -> floats.

**Review by delihus — CHANGES_REQUESTED** [review]
A couple suggestions to the descriptions and a request to change floats values in the managers to double or use `static_cast`.

File: husarion_ugv_manager/config/safety_manager_parameters.yaml [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+        type: int
+        default_value: 6
+        description: Moving average window length used to smooth out temperature readings of the battery.
+        validation: { gt<>: 0 }
+
+  bt_project_path:
+    type: string
+    default_value: ""
+    description: Path to a BehaviorTree project.
+    validation: { not_empty<> }
```
Comment (delihus) [inline husarion_ugv_manager/config/safety_manager_parameters.yaml]: This validation fails the tests. Please update tests or remove this validation.

File: husarion_ugv_manager/config/lights_manager_parameters.yaml [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+          type: double
+          default_value: 0.4
+          description: If the Battery percentage drops below this value, the animation indicating a low Battery state will start being displayed.
+          validation: { bounds<>: [0.0, 1.0] }
+
+  bt_project_path:
+    type: string
+    default_value: ""
+    description: Path to a BehaviorTree project.
+    validation: { not_empty<> }
```
Comment (delihus) [inline husarion_ugv_manager/config/lights_manager_parameters.yaml]: This validation fails the tests. Please update tests or remove this validation.

```bash
    C++ exception with description "Invalid value set during initialization for parameter 'bt_project_path': Parameter 'bt_project_path' cannot be empty" thrown in the test fixture's constructor.
```

**Review by delihus — CHANGES_REQUESTED** [review]

File: husarion_ugv_battery/src/battery_driver_node.cpp (hunk 1 above)
Comment (ghost) [reply husarion_ugv_battery/src/battery_driver_node.cpp:86]: I would prefer to have an additional cast here and not rely only on validation to discard negative parameter values

File: husarion_ugv_manager/src/lights_manager_node.cpp [hunk 21]
```diff
@@ -69,7 +70,7 @@ void LightsManagerNode::Initialize()
     "hardware/e_stop", rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable(),
     std::bind(&LightsManagerNode::EStopCB, this, _1));
 
-  const float timer_freq = this->get_parameter("timer_frequency").as_double();
+  const float timer_freq = this->params_.timer_frequency;
```
Comment (delihus) [inline husarion_ugv_manager/src/lights_manager_node.cpp]:
```suggestion
  const double timer_freq = this->params_.timer_frequency;
```

like in the safety_manager

**Review by delihus — CHANGES_REQUESTED** [review]

**Review by delihus — APPROVED** [review]

**Pull request merged**